teps/0084-endtoend-provenance-collection.md

```diff
@@ -188,13 +129,254 @@ The `in-toto` format created by chains wraps the [SLSA Provenance v0.2](https://
+            },
+            {
+              "name": "IMAGE_URL",
+              "value": "registry.example.com/minimal-container/min:latest\n"
+            }
+          ]
+        },
+        {
+          "name": "image-security-scan",
+          "ref": {
+            "name": "trivy-scanner",
+            "kind": "Task",
+            "bundle": "gcr.io/tekton-releases/catalog/upstream/trivy-scanner@sha256:e4c2916f25ce2d42ec7016c3dc3392e527442c307f43aae3ea63f4622ee5cfe4"
+          },
+          "startedOn": "2022-07-06T17:17:43Z",
+          "finishedOn": "2022-07-06T17:17:55Z",
+          "status": "Succeeded",
+          "steps": [
+            {
+              "entryPoint": "#!/usr/bin/env sh\n  cmd=\"trivy $* $(params.IMAGE_PATH)\"\n  echo \"Running trivy task with command below\"\n  echo \"$cmd\"\n  eval \"$cmd\"\n",
+              "arguments": [
+                "$(params.ARGS)"
+              ],
+              "environment": {
+                "container": "trivy-scan",
+                "image": "docker.io/aquasec/trivy@sha256:dea76d4b50c75125cada676a87ac23de2b7ba4374752c6f908253c3b839201d9"
+              },
+              "annotations": null
+            }
+          ],
+          "invocation": {
+            "configSource": {},
+            "parameters": {
+              "ARGS": "[\"image\"]",
+              "IMAGE_PATH": "\"$(tasks.image-build.results.IMAGE_URL)\"",
+              "TRIVY_IMAGE": "\"docker.io/aquasec/trivy@sha256:dea76d4b50c75125cada676a87ac23de2b7ba4374752c6f908253c3b839201d9\""
+            }
+          }
+        }
+      ]
+    },
+    "metadata": {
+      "buildStartedOn": "2022-07-06T17:17:12Z",
+      "buildFinishedOn": "2022-07-06T17:17:55Z",
+      "completeness": {
+        "parameters": false,
+        "environment": false,
+        "materials": false
+      },
+      "reproducible": false
+    },
+    "materials": [
+      {
+        "uri": "git+https://github.com/lcarva/minimal-container.git",
+        "digest": {
+          "sha1": "a5bdfa264ac87bb2bec104ef9ddf2207b5b210c9"
+        }
+      }
+    ]
+  }
+}
+```
+
+</details>
+
 
 ##### 3. Attestation Format
-(As an optimizion option) Instead of creating seperate attestation records for `taskrun`, `pipelinerun`, `event-payload`, create a single attestation record at the "end" of a `pipelinerun` that includes everything.
+(As an optimization option) Instead of creating separate attestation records for `taskrun`, `pipelinerun`, `event-payload`, create a single attestation record at the "end" of a `pipelinerun` that includes everything.
 
 In our running example above, with this changes, for a given `image` we can attest that:
-1.  A pipeline was trigger by event with attested payload
-2. In the pipeline, 3 tasks were execure in this order: "git-clone" --> "security-scan" --> "image-build".
+1. A pipeline was trigger by event with attested payload
+2. In the pipeline, 3 tasks were executed in this order: "git-clone" --> "security-scan" --> "image-build".
```

Comment:
If the artifact generation task failed i.e. "image-build" in this example, will the pipelinerun attestation still be generate?

Reply:
That's a good question. Today, Chains generates attestations for failed TaskRuns, so I'd like attestations for PipelineRuns to behave the same way.  There's an issue filed questioning the existing behavior of TaskRuns: https://github.com/tektoncd/chains/issues/484

Reply:
Thanks for the pointer!